### PR TITLE
Updated Coding Standard release proposal

### PR DESCRIPTION
**Pull request by igorin**
Included cadence, side effects, and notifications updates discussed today

## Problem

We still had some unknowns regarding cadence and release notes.

## Solution

Monthly release cadence, no breaking changes, changes in priorities and scoring synchronized with M2 patch releases, changes in test set and fail conditions synchronized with M2 minor releases.

## Requested Reviewers

@lenaorobei @antonkril

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/architecture?pullRequest=167) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/architecture?pullRequest=167) before we can accept your contribution.<br/>

File: design-documents/testing/static/coding-standard-versioning.md [hunk 1]
```diff
@@ -25,12 +25,21 @@ Since we want all users (core developers, community contributors, system integra
         ...
     }
 ```
+## Release cadence
+New versions will be released monthly (if there are changes since the last release), as well as with each Magento patch/minor release. They will include all the latest changes to the checks. There will be three distinct cases that will differentiate what changes they introduce:
```
Comment (lenaorobei) [inline design-documents/testing/static/coding-standard-versioning.md]: That sounds good to me, however we do not cover here critical bug fixes and how we will deliver it.
I suggest adding something like:

- Critical bugfixes may be delivered earlier to unblock affected teams.

Comment (igorin) [reply design-documents/testing/static/coding-standard-versioning.md]: Would this be a better description?

Cadence plan:
	1.	Critical fixes (broken suite, broken sev-10 tests) deployed as ready in isolation 
	2.	Monthly releases that include:
	a.	Bugfixes
	b.	New non-failing tests (warning only)
	c.	(future) no changes in code quality scores – if new tests have been added, they need to be excluded from scoring
	3.	Magento-bundled releases that include:
	a.	Patch releases: all the above + non-failing (sev 9 and below) severity promotions/demotions, and changes to scoring
	b.	Minor releases: all the above + new failing tests can be introduced

**Comment by orlangur** [comment]
@igorin why so complicated? Why not use the approach I described during the call?

**Comment by igorin** [comment]
@orlangur can you write down your proposal? unfortunately, it looks like I missed that particular call

the strategy above is an intelligent compromise between the architecture team, the development team, and the product team, taking into account dependencies and limitations such as:
- Managing developers' expectations
- Communicating breaking changes well in advance
- Allowing developers to plan their technical debt management activities
- Allowing engineering managers to plan releases with concrete expectations regarding coding standard updates

This strategy will impact all internal development teams, Community Engineering, extension developers, VBE partners, and generally any developer who is integrating coding standard checks into their production deployment pipeline.

File: design-documents/testing/static/coding-standard-versioning.md [hunk 2]
```diff
@@ -25,12 +25,40 @@ Since we want all users (core developers, community contributors, system integra
...
 
 ## Side Effects
 - No `patch` or `minor` dependency will be possible, so any change may lead to build (code check) failure.
 - If released Magento package will contain `"*"` dependency any update may lead to build (code check) failure.
+- Developers will need to track changes in the project either by paying attention to notifications or by following the project in GitHub, in order to make their "in development" code adherent to the upcoming coding-standard release. This will be especially important for patch releases, and critical for minor releases.
 
 ## Marketplace Notifications
+- A week in advance before each release, a notification containing the upcoming changes will be sent. Additional bugfixes may be included into the release after this date, but all possible effort will be applied to not introducing new checks after the notification had been sent. The notification will inform of the entire scope of release changes: bug fixes, severity/scoring changes, and new checks.
```
Comment (ravmenon) [inline design-documents/testing/static/coding-standard-versioning.md]: For quarterly releases with severity 10s', we will need to give more notice, say a month,  to give developers time to make the fixes.  This means at least a month before the quarterly release.  Any issues found during this period (say a new severity 10 rule), will happen for the next release train, which will be another quarter away,.

**Pull request merged**